### DISCUSSION TITLE
What's the future of Servo?

### DISCUSSION THREAD

**Discussion by jessaleks** [servo/servo#27575] [General] [answered]
Is the project being killed off?

Reportedly the team is gone https://twitter.com/directhex/status/1293352458308198401?s=19

**Comment by glasscubedev** [comment]
If so, we need to build a new browser engine. I cannot think that the gateway to the internet will only be available through Chrome. Just fork it and start again. I cannot let Servo die.

**Reply by Mubelotix** [reply]
We must unite. We have to continue the project

**Reply by rbanffy** [reply]
Gecko is still around. IIRC, a lot of the optimizations discovered due to work in Servo have guided its evolution to its present state.

**Reply by niutech** [reply]
Why to build a new web engine rather than just continue to develop Servo as a community project? It isn't archived, so collaborators [will continue to merge PRs](https://github.com/servo/servo/discussions/27584#discussioncomment-51146). This is the power of open source!

As for Chrome-only web, there are alternative web engines apart from Webkit/Blink and Gecko/Servo: NetSurf, Ultralight, Sciter.

**Reply by skyfaller** [reply]
> As for Chrome-only web, there are alternative web engines apart from Webkit/Blink and Gecko/Servo: NetSurf, Ultralight, Sciter.

Ultralight is only partially open source, and it is based on Webkit. Sciter is proprietary. NetSurf is fun, I enjoy it, but it doesn't support Javascript which will hinder mainstream adoption.

**Reply by niutech** [reply]
@skyfaller JS support in Netsurf is still being [developed](http://source.netsurf-browser.org/netsurf.git/log/content/handlers/javascript) and is available since Netsurf 3.4, although disabled by default. Maybe those of you who know C could help developing it further?

**Reply by jessaleks** [reply]
Not supporting JS isn't that bad, actually 😈

**Comment by rubyFeedback** [comment]
People need to be less reluctant to fork. Not even in regards to Servo alone, but more generally so - browsers are gate entry points to information. There are way too many corporations already involved with a separate agenda (decoupled from end users).

I don't think one has to adhere to the W3C ("everyone must embrace DRM") to assume there is a single entity dictating on mankind which standards everyone has to adhere to; or Google for that matter. Other projects such as libreoffice show that you can be successful with a large code base too. Mozilla looked elsewhere years ago already since they conceded defeat ("we can't win against Google"). Now would be an EXCELLENT time to see that a broader movement could get together - not just in regards to any particular engine alone, but the whole www as it is. (Ideally a decentralized variant of the www, but that's my personal opinion - just chiming in to give people a nudge.)

**Reply by tmccombs** [reply]
Much as I hate it, a browser that doesn't support EME/DRM will have an even harder time getting adoption, since basically every (legal) streaming service requires it. And multiple projects have shown how difficult it is to get a license for Widevine in a forked browser.

**Reply by Ajedi32** [reply]
> People need to be less reluctant to fork. Not even in regards to Servo alone, but more generally so 

While I fully agree with this, I believe the the main issue in this particular case is that it takes a _lot_ of resources to maintain a browser engine. If you can find a half-dozen or so skilled and dedicated volunteers, or raise enough money to pay people to work on it full time for the foreseeable future then great! I say go for it! That's by no means a trivial task though.

**Reply by foucist** [reply]
> Much as I hate it, a browser that doesn't support EME/DRM will have an even harder time getting adoption, since basically every (legal) streaming service requires it.

I don't think it's unreasonable to use a separate browser specifically for netflix/etc.  The daily driver browser doesn't really need EME/DRM.  We should be willing to split these things up into two separate apps.

**Reply by tmccombs** [reply]
> I don't think it's unreasonable to use a separate browser specifically for netflix/etc.

Maybe for a small number of tech-savvy users. But the majority of people are not going to be willing to use a different browser for different things. If you just want to make a niche browser, sure go ahead. If you want a browser that can pick up enough market share that it can have any say in web standards, it will need to Just Work for all the things that the average user does on the web. And ideally, do them better than the current dominate browser. And my hope for a servo based browser, whether that is firefox or something else, is the latter.

**Reply by jessaleks** [reply]
> > Much as I hate it, a browser that doesn't support EME/DRM will have an even harder time getting adoption, since basically every (legal) streaming service requires it.
> 
> I don't think it's unreasonable to use a separate browser specifically for netflix/etc. The daily driver browser doesn't really need EME/DRM. We should be willing to split these things up into two separate apps.

It is unreasonable to use a separate browser. Being a niche browser for the privacy-conscious crowd doesn't really work for Mozilla, does it?

It's not that Chrome is better, faster, etc (which is disputable). It's that Chrome is associated with being the de facto standard web browser. 

It's always possible to beat any competitor, question is, how hard is it?

**Reply by munael** [reply]
> > I don't think it's unreasonable to use a separate browser specifically for netflix/etc.
> 
> Maybe for a small number of tech-savvy users. But the majority of people are not going to be willing to use a different browser for different things. If you just want to make a niche browser, sure go ahead. If you want a browser that can pick up enough market share that it can have any say in web standards, it will need to Just Work for all the things that the average user does on the web. And ideally, do them better than the current dominate browser. And my hope for a servo based browser, whether that is firefox or something else, is the latter.

Possible future path: _One_ open source project works on providing a very lean and streamlined embeddable browser with no dedicated UI of its own, and _that_ mini-browser gets the license.

All other interested OSS browsers integrate that DRM-enabled mini-browser as an external app that they invoke somehow in a window provided by them, but everything it does internally is its own.

It could then provide various hooks to enable the host browser to manage its artifacts, like cookies, certificates, and so on.

**Reply by shakeyourbunny** [reply]
SeaMonkey could take up the slack, it just needs some modernization.

**Comment by ghost** [comment]
Hey!

* i am sure people are very shocked by this decision of mozilla. Which means this is potential to mobilize people.
* based on this huge mobilization potential, we could start a fundraiser to maintain a server/firefox/whatever browser.
* this effort could be supported by [nlnet.nl](https://nlnet.nl). that is a foundation who has in the past financed many free software projects. The list of funded projects is available [here](https://nlnet.nl/project/current.html).

in short, i think **many** people are willing to donate for a maintained firefox fork that works for many people.

**Reply by jessaleks** [reply]
It doesn't make sense to finance a fork of Firefox labeled as such. It doesn't make sense to finance something similar, even.

The core could remain the same, sure. It only makes sense to ship something that aims to solve common people's problems.

**Reply by ghost** [reply]
Here's the thing: we *could* do this, we *could* do this other thing...

There's no hope for Servo if big corporations don't fund it. And no one is willing to do so. Mozilla is focusing on survival right now, not on creating a new web engine. I'd have loved to see Servo production-ready, as its embedding capabilities would have made a lot of people adopt it and democratize the web market a *little* bit. But let's be realistical, a miracle has to happen...

**Comment by shakeyourbunny** [comment]
There is something called UXP, they surely need some more software engineers to lift it up.

Another option is to help the SeaMonkey guys to rise back like a phoenix from the ashes (*), but they are in dire need people who lend them a hand or two. This would also help Thunderbird to survive.

(*) pun intended. For the younger folk: Firefox was a fork of something that was the predecessor of SeaMonkey and originally called Phoenix.

It is really sad that this decision was made to lay off people who were working on the core future of Firefox.

**Comment by wilhelmy** [comment]
This really makes me wonder how decisionmaking within Mozilla works and how things are prioritized... I'd donate money for Firefox and its core technologies (wouldn't be much, but I'd be ok with a yearly subscription) but I wouldn't want to donate money through a Mozilla that makes decisions in an haphazard way, apparently after losing funding from Google when it was clear for years that this might happen eventually, and I'd expect transparency and more forethought from any kind of sustainable successor organization.

Then again, there is no guarantee that donations can ensure Firefox's continued existence, which ties into the whole "how do we fund FOSS at all if not for google playing patron until they don't" issue. From a financial standpoint and for tightening their grip on web standards it probably makes sense to cut Mozilla off at this point.

I'm sorry you lost your jobs and I appreciate your work on Servo.

Edit: reading https://www.theregister.com/2020/08/14/mozilla_google_search/ it seems like they didn't lose funding after all, which makes it all the more mysterious.

**Reply by stefee** [reply]
Quote from the article:

> Updated to add on August 14
> A Moz spokesperson has been in touch to remind us that the Servo team's work made its way into Firefox in various forms, and to assure us Rust will be used to develop other parts of the browser.
> 
> "The Servo research project proved that Rust was an excellent language for browser development," the spokesperson said, "and incubated a number of important technologies that are now part of core Firefox, such as the style system and our new lighting fast GPU-based rendering system, WebRender.
> 
> "Going forward, we will be writing many new browser components in Rust directly in Firefox rather than in a separate project."

**Reply by stefee** [reply]
So it would seem that although Mozilla has parted ways with the Servo project, it will still develop Rust browser engine components in FireFox directly (perhaps forked from Servo?).

It is very unclear to me how motivated Mozilla are to do this work seeing as Mozilla has just let go of most if not all of the people that were developing browser engine components in Rust.

**Reply by ghost** [reply]
I don't know you, but statistically speaking, would *you* donate money? I don't think so. If yes, why are you not doing so right now?

**Comment by Toasterson** [comment]
I would caution about acting too fast. It has been a problem, that companies do not care about things that do not make money. But that does not mean they kill things that are part of their core product off at a whim.

Forking blindly will for sure kill this project off. I have seen this happening before with the leftovers of Sun Microsystems. We forked too happily. Did not align with a big enough community to make an impact. We are changing now but we are in a much worse position than if we did a unifying talk in the beginning. I agree that a company will not care about servo. There must be a structure that cares about pushing servo. And that structure will need a sustenance that allows developers to work at least part time on servo. Joining other groups means joining their cause not the cause of furthering servo.

Whatever ends up happening. For servo to survive requires a couple of basics that need to be met. From my experience those are:
- Dedication of people to a common and shared Vision (Not companies, people)
- A structure that allows developers to make time for this. (Mostly talking about compensation for invested time) (everybody has a live that needs to be financed)
- A source of sustenance (Survival cashflow. No need for earnings and no need for growth)
- Options for more usage outside of the one product (Make an engine not a complete browser, allow that engine to be used in other use cases. i.e. headless webpage parsing)

I hope for servo to succeed and I wanted to use it for the headless/parser use case and I still consider it a valid option for that.

In Short. Dedicate yourself to this cause. Do not dedicate yourself to another cause. Go into other Companies and find a way to use servo there.

**Comment by paulrouget** [comment]
Just then you know: lot of us do not have the mental bandwidth, at the moment, to think or plan anything for the future of Servo. The news came to us as a surprise. It's not just about the project, but also our jobs.

Not sure when and who will speak about the future of the project, but maybe do not expect anyone to come up with answers right now.

**Reply by Toasterson** [reply]
You have my fullest support. Take the time you need. All of you.

Thank you very much for the answer. Knowing you are still there is already a lot.

**Reply by resynth1943** [reply]
Well, as much as I don't respect Mozilla, I'd definitely love to see the Servo project continue. Perhaps it could also be continued under a separate foundation... If Servo and Gecko were to suddenly die, we would only be left with Blink and WebKit. While I do love WebKit, there should definitely be more options for people who don't want to support corporations like Google, by using their various engines.

Thank you for your prompt answer. I wish you all the best in the future. ❤️

**Reply by justicezyx** [reply]
My support is with you Paul!

Servo project represents significant value to the Web. Give yourself time to rest. It's best that you and the team can have the peace of mind to come up with the best plan to continue the project.

**Reply by archseer** [reply]
It would be interesting to try and set up some type community funding/Github Sponsor program. Maybe enough funding could be raised that way for the team to continue?

**Comment by justicezyx** [comment]
Who are the people affected from Servo team? I am interested in learning their LinkedIn profile, where there might be opportunities around, and they can continue to work on Servo.

**Reply by poovamraj** [reply]
I hope this helps

https://mozillalifeboat.com/

https://talentdirectory.mozilla.org/

**Comment by jason-oliveira** [comment]
The only reason you would have to fork is if this repo stops accepting pull requests.

Otherwise, the project is active, and everyone posting in here should continue (or begin) submitting pull requests.

**Comment by EdorianDark** [comment]
One possibility would be to use "github Sponsor" or something similar.

I would be willing to give 1000€ per month for an year, if it helps to push Parthfinder and maybe even Layout 2020 into Firefox.
I don't know it is realistically, but I would really like to be able to embed Servo easily like Sciter or CEF.

**Reply by SimonSapin** [reply]
See https://servo.org/#support-and-donations

**Comment by Bug-Reaper** [comment]
Lol killed off just like positron... bastards. Still seems very active from github repo, I'll be PRing for this soon I'm sure.

**Comment by arxo** [comment, marked as answer]
Looks like Servo moved over to the Linux Foundation:
https://blog.servo.org/2020/11/17/servo-home/

**Comment by Announcement** [comment]
i’m not too familiar with all of this but i know the main thing that keeps bringing me back to the servo project is the idea that one day it could be used to build applications. i was more hoping servo would be a cross platform interface sort of like electron but with support for like android or apple mobile operating systems as well potentially. i feel like a lot of us would be on board and adaptation would expand vastly and rapidly.

**Reply by BasixKOR** [reply]
Tauri has been exploring integrating Servo (and other Rust-based web components) into their project, and you might want to join their Discord to see the progress ;)
